schemas/context/namespace.schema.json

```diff
@@ -0,0 +1,37 @@
+{
+   "meta:license":[
+      "Copyright 2017 Adobe Systems Incorporated. All rights reserved.",
+      "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+      "you may not use this file except in compliance with the License. You may obtain a copy",
+      "of the License at https://creativecommons.org/licenses/by/4.0/"
+   ],
+   "$id":"https://ns.adobe.com/xdm/context/namespace",
+   "$schema":"http://json-schema.org/draft-06/schema#",
+   "title":"Namespace",
+   "type":"object",
+   "description":"The Datasource acts as a namespace or unique identifier associated with a collection of data. Each EndUserID is associated with a given Datasource. A user can create and obtain information about a Datasource during the solution onboarding.\n",
+   "definitions":{
+      "namespace":{
+         "properties":{
+            "xdm:id":{
+               "title":"Identifier",
+               "type":"integer",
```

Comment:
Numeric IDs are a bad idea. Give the poor Namespace a proper URI, which might end with an integer for now.

Reply:
That's fine, but should go to the system that generates and provides those id's and have them produce the full URI?

Reply:
For the data platform, I'd assume the system that manages namespaces and the system that generates IDs/URIs is identical. In ingestion, you could even go so far to ignore data that belongs to an unknown namespace, i.e. that has a namespace URI that is outside of your known URI schema.

Reply:
The efficiency and scale of this means that integer processing vs processing a URI is important.

Reply:
Hashing a string into an integer is a really fast operation (especially considering that you are in the context of JSON processing). 

But keeping a list of integers centrally managed and unique over decades is almost impossible.

Reply:
"(especially considering that you are in the context of JSON processing)" The schema is represented in JSON, the data is sometimes in some cases, but mostly the industry and customer implementations keep the data is a binary format, db logs, parquet and process it from that format without ever going to JSON.

If a different "governor of data" wants to use a different option to generate the integers they are free to do so. That is an implementation detail.

What we are proposing here is implemented at scale and in production for many years.

Reply:
We should define the IDs as URIs.  Any ID (numerical or string) can be made a URI.  We should not put any restrictions on the form of the URI though, so that implementors have maximum flexibility.

Reply:
@cdegroot-adobe the representation that XDM specifies and that we discuss is JSON. Any system processing it, should represent data in a way that is convenient and efficient, by any means. 

In this case here, where the URIs is not restricted in any way and will not be interpreted by your system, hashing it can be an efficient way to turn it into a fixed-length number.

Reply:
Action item for @jwen-adobe:
- [ ] rename to `@id`
- [ ] make it a `uri`

Reply:
@trieloff @cmathis done.